Source/NSString.m

```diff
@@ -6765,3 +6775,309 @@ - (void) setString: (NSString*)aString
 
 @end
 
+/**
+ * An item in a .stringsdict file.
+ * https://developer.apple.com/documentation/xcode/localizing-strings-that-contain-plurals
+ */
+@implementation GSLocalizedString
+
+- (id)initWithDictionary:(NSDictionary *)dict defaultValue:(NSString *)value
+{
+  _dict = dict;
+  return [self initWithString:value];
+}
+
+/* Returns a format that's appropriate for the pluralization of the arguments in
+ * the current language */
+- (NSString *)getPluralizedFormat:(va_list)argList
+{
+  /* Sample dictionary:
+      NSStringLocalizedFormatKey=%#@x_hours@, %#@x_minutes@
+      x_hours=NSDictionary {
+	NSStringFormatSpecTypeKey=NSStringPluralRuleType
+	NSStringFormatValueTypeKey=ld
+	zero=0 hours
+	one=1 hour
+	other=%1$ld hours
+      }
+      x_minutes=NSDictionary {
+	NSStringFormatSpecTypeKey=NSStringPluralRuleType
+	NSStringFormatValueTypeKey=ld
+	zero=0 minutes
+	one=1 minute
+	other=%2$ld minutes
+      }
+   */
+
+  NSString *formatKey = _dict[@"NSStringLocalizedFormatKey"];
+  if (!formatKey)
+    {
+      return self;
+    }
+
+  static dispatch_once_t      onceToken;
+  static NSRegularExpression *regex = nil;
+
+  regex = [NSRegularExpression
+    regularExpressionWithPattern:@"%#@([^@]+)@"
+			 options:NSRegularExpressionCaseInsensitive
+			   error:nil];
+
+  NSArray<NSTextCheckingResult *> *matches =
+    [regex matchesInString:formatKey
+		   options:0
+		     range:NSMakeRange(0, [formatKey length])];
+
+  va_list args;
+  va_copy(args, argList);
+
+  NSMutableString *newFormat = [formatKey mutableCopy];
+  NSInteger	   offset = 0;
+
+  for (NSTextCheckingResult *match in matches)
+    {
+      NSString *varKey = [formatKey substringWithRange:[match rangeAtIndex:1]];
+
+      NSDictionary<NSString *, NSString *> *varDict = _dict[varKey];
+
+      // Get the numeric value of the variable (luckily, only "%ld", "%lu", and
+      // "%1g" is used).
+      unsigned long number;
+      NSString	   *type = varDict[@"NSStringFormatValueTypeKey"];
+
+      if ([type isEqualToString:@"ld"])
```

Comment:
If we're adding these anyway, can we also add
* `%d` -> `int`
* `%u` -> `unsigned`

while we're at it? While we don't use these right now, somebody else (or us in the future) may end up with them